bot/exts/info/patreon.py

```diff
@@ -0,0 +1,162 @@
+            "[Support us on Patreon](https://pydis.com/patreon)"
+        )
+
+        sending_channel = utils.channel.get_or_fetch_channel(constants.Channels.meta)
+
+        await sending_channel.send(
+            embed=discord.Embed(
+                description=message,
+                colour=colour
+            )
+        )
+
+    async def send_current_supporters(self, channel: discord.TextChannel) -> None:
+        """Send the current list of patreon supporters, sorted by tier level."""
+        await self.bot.wait_until_guild_available()
+
+        guild = self.bot.get_guild(constants.Guild.id)
+
+        tier_1_or_above_patrons = set(guild.get_role(constants.Roles.patreon_tier_1).members)
+        tier_2_or_above_patrons = set(guild.get_role(constants.Roles.patreon_tier_2).members)
+        tier_3_patrons = set(guild.get_role(constants.Roles.patreon_tier_3).members)
+
+        tier_1_patrons = tier_1_or_above_patrons - tier_2_or_above_patrons - tier_3_patrons
+        tier_2_patrons = tier_2_or_above_patrons - tier_3_patrons
+
+        tier_1_patrons = {f"{patron.mention} ({patron.name}#{patron.discriminator})" for patron in tier_1_patrons}
+        tier_2_patrons = {f"{patron.mention} ({patron.name}#{patron.discriminator})" for patron in tier_2_patrons}
+        tier_3_patrons = {f"{patron.mention} ({patron.name}#{patron.discriminator})" for patron in tier_3_patrons}
+
+        embed_list = []
+
+        embed_patron_info = discord.Embed(
+            title="Patreon Supporters",
+            description=(
+                "We use the money from Patreon to offer excellent prizes for all of our events. Prizes like t-shirts, "
+                "stickers, microcontrollers that support CircuitPython, or maybe even a mechanical keyboard.\n\n"
+
+                "You can read more about how Patreon supports us, or even support us yourself, on our Patreon page "
+                "[here](https://pydis.com/patreon)!"
+            )
+        )
+
+        embed_list.append(embed_patron_info)
+
+        if tier_1_patrons:
+            embed_tier_1 = discord.Embed(
+                title="Tier 1 patrons",
+                description="\n".join(tier_1_patrons),
+                colour=guild.get_role(constants.Roles.patreon_tier_1).colour
+            )
+            embed_list.append(embed_tier_1)
+
+        if tier_2_patrons:
+            embed_tier_2 = discord.Embed(
+                title="Tier 2 patrons",
+                description="\n".join(tier_2_patrons),
+                colour=guild.get_role(constants.Roles.patreon_tier_2).colour
+            )
+            embed_list.append(embed_tier_2)
+
+        if tier_3_patrons:
+            embed_tier_3 = discord.Embed(
+                title="Tier 3 patrons",
+                description="\n".join(tier_3_patrons),
+                colour=guild.get_role(constants.Roles.patreon_tier_3).colour
+            )
+            embed_list.append(embed_tier_3)
```

Comment:
I was thinking about a way to do this, and think I've come up with a way to do it.

additionally, I think that tier 3 should be before tier 2 and 1, as tier 3 is above tier 2.

Using a dictionary we can match the roles to the members, which ends up like this:
*note: this builds off of storing the patreon roles in a list*
```py
embeds = []
for num, role in reversed(enumerate(patreon_role_list)):
    if not role.members:
       # skipping this role since there are no members
       continue

    # right here the description could be formulated, instead of above

    embeds.append(discord.Embed(
                title=f"Tier {num + 1} patrons",
                description="\n".join(role.members),
                colour=role.colour
            )
    )

if not embeds:
   # there's no embeds, so there's no embed to send
   # yes, this is an extreme edge case to not have any embeds, but should be prepared for it anyhow.

```